### ISSUE TITLE
Harpy crashes in alertTypeForVersion if updating from 6.0 -> 6.0.1(in our case)

### ISSUE THREAD

**Issue by FabianKS** [ArtSabintsev/Harpy#107]
There is no check if the old version has as many components as the new version. 
In our case Harpy crashed on accessing `oldVersionComponents[2]`.

```
- (void)alertTypeForVersion:(NSString *)currentAppStoreVersion
{
    // Check what version the update is, major, minor or a patch
    NSArray *oldVersionComponents = [[self currentVersion] componentsSeparatedByString:@"."];
    NSArray *newVersionComponents = [currentAppStoreVersion componentsSeparatedByString: @"."];

    BOOL oldVersionComponentIsProperFormat = (2 <= [oldVersionComponents count] && [oldVersionComponents count] <= 4);
    BOOL newVersionComponentIsProperFormat = (2 <= [newVersionComponents count] && [newVersionComponents count] <= 4);

    if (oldVersionComponentIsProperFormat && newVersionComponentIsProperFormat) {
        if ([newVersionComponents[0] integerValue] > [oldVersionComponents[0] integerValue]) { // A.b.c.d
            if (majorUpdateAlertType) alertType = _majorUpdateAlertType;
        } else if ([newVersionComponents[1] integerValue] > [oldVersionComponents[1] integerValue]) { // a.B.c.d
            if (minorUpdateAlertType) alertType = _minorUpdateAlertType;
        } else if ((newVersionComponents.count > 2) && ([newVersionComponents[2] integerValue] > [oldVersionComponents[2] integerValue])) { // a.b.C.d
           if (patchUpdateAlertType) alertType = _patchUpdateAlertType;
        } else if ((newVersionComponents.count > 3) && ([newVersionComponents[3] integerValue] > [oldVersionComponents[3] integerValue])) { // a.b.c.D
            if (revisionUpdateAlertType) alertType = _revisionUpdateAlertType;
        }
    }
}```

**Comment by jkandzi** [comment]
should be fixed by #108

**Comment by ArtSabintsev** [comment]
Fixed in Harpy v3.4.5.
